### PR TITLE
[Enhencement](env) Checking Master branch must use JDK17

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number: close #xxx

Add to check the JDK version in `env.sh`,  and force master to use java 17 version

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

File: env.sh [hunk 1]
```diff
@@ -213,20 +260,41 @@ if test -z "${BUILD_THIRDPARTY_WIP:-}"; then
...
         JAVA="${JAVA_HOME}/bin/java"
         JAVAP="${JAVA_HOME}/bin/javap"
+
+        echo "Check JAVA_HOME version"
+        java_version="$(
+            set -e
+            jdk_version "${JAVA}"
+        )"
+        if [[ "${java_version}" -ne 17 ]]; then
+            echo "JAVA_HOME does not point to JDK-17."
```
Comment (morningman) [inline env.sh:283]: print the $JAVA_HOME for easy debug

File: env.sh [hunk 2]
```diff
@@ -213,20 +260,41 @@ if test -z "${BUILD_THIRDPARTY_WIP:-}"; then
 
     # check java home
     if [[ -z "${JAVA_HOME}" ]]; then
-        JAVA="$(command -v java)"
-        JAVAP="$(command -v javap)"
+        if [[ "${JDK_17}" ]]; then
+            echo "Use JDK_17"
```
Comment (morningman) [inline env.sh:264]: print the $JDK_17 for easy debug

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by hello-stephen — APPROVED** [review]
LGTM

**Pull request merged**